### PR TITLE
exa: remove as unmaintained upstream

### PR DESCRIPTION
**Pull request by oxalica**
## Description of changes

https://github.com/ogham/exa/issues/1243

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
Looks reasonable.

**Review by teto — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Comment by peterhoeg** [comment]
Could we put in a compatibility symlink `exa -> eza` so that things don't break if people swap the packages?

**Comment by peterhoeg** [comment]
I mean a symlink to the binary.

**Comment by teto** [comment]
I find it more confusing than anything else, this should be up to users (else upstream would have relinquished the name)

**Comment by peterhoeg** [comment]
Exa upstream has allegedly gone AWOL. You could then argue that eza upstream should have a flag for putting in a compatibility symlink, but in order to follow the principle of least surprise, we could then handle it downstream.

**Comment by cafkafk** [comment]
> You could then argue that eza upstream should have a flag for putting in a compatibility symlink

Hi, I'm eza upstream. I'm not sure how such a flag would work, but feel free to create a feature request outlining this, or perhaps a PR.

**Comment by oxalica** [comment]
I agree with @teto that it's confusing to alias one project to a fork maintained by a different group of people, even though the original author endorsed it. End-users should acknowledge this and then decide which one to use, or probably go back to coreutils.

**Comment by cafkafk** [comment]
> maintained by a different group of people.

For the record, Mélanie Chauvel maintained `exa` for years and has been active in `exa` since 2018, and she is also a maintainer on `eza`.

I was made aware that [Archlinux Extra](https://archlinux.org/packages/extra/x86_64/eza/) does have [symlink](https://gitlab.archlinux.org/archlinux/packaging/packages/eza/-/blob/main/PKGBUILD#L44) from `exa` to `eza`. There is also a noticeable amount of users that have asked for this.

That said, I personally prefer users making this a conscious choice.

But we might provide some feature flag that creates a symlink or something like it, and I'll make sure to add it to `eza` in [nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/tools/misc/eza/default.nix#L49) on the following release as a `lib.optional`, in the same way we currently do `gitSupport`. That *should* make most people happy. (@peterhoeg)?

**Comment by oxalica** [comment]
> > maintained by a different group of people.
> 
> For the record, Mélanie Chauvel maintained `exa` for years and has been active in `exa` since 2018, and she is also a maintainer on `eza`.

Okay, that's reasonable.

> But we might provide some feature flag that creates a symlink or something like it, and I'll make sure to add it to `eza` in [nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/tools/misc/eza/default.nix#L49) on the following release as a `lib.optional`

We can definitely do that, probably just in `installPhase`. An override-able parameter `exaAlias` could be fine, as we already have `{vi,vim}Alias` for `neovim` [here](https://github.com/NixOS/nixpkgs/blob/9699c0636d4beda82205295465a89bf831c8813e/pkgs/applications/editors/neovim/wrapper.nix#L98). Since `exa` is already removed, having a `true` default seems fine to me.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -496,6 +496,7 @@ mapAliases ({
   euca2tools = throw "euca2ools has been removed because it is unmaintained upstream and still uses python2"; # Added 2022-01-01
   evilvte = throw "evilvte has been removed from nixpkgs for being unmaintained with security issues and dependant on an old version of vte which was removed"; # Added 2022-01-14
   evolution_data_server = throw "'evolution_data_server' has been renamed to/replaced by 'evolution-data-server'"; # Converted to throw 2022-02-22
+  exa = throw "'exa' has been removed because it is unmaintained upstream. Consider using 'eza', a maintained fork"; # Added 2023-09-07
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:499]: Why not
```suggestion
  exa = eza; # Added 2023-09-07
```
?

**Comment by dotlambda** [comment]
> else upstream would have relinquished the name

They didn't say they wouldn't because they didn't say anything.

`eza` is clearly a continuation of `exa` so the latter should be an alias.

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (wegank) [reply pkgs/top-level/aliases.nix:499]: For me, this was not acceptable before #254600. Now it is.

**Comment by cafkafk** [comment]
I would like if you could at least keep a trace telling users that they're not installing exa

**Comment by teto** [comment]
> An override-able parameter exaAlias could be fine, as we already have {vi,vim}Alias for neovim [here](https://github.com/NixOS/nixpkgs/blob/9699c0636d4beda82205295465a89bf831c8813e/pkgs/applications/editors/neovim/wrapper.nix#L98)

and I find that weird too xD neovim is neither vim or vi so it's odd/confusing to create those aliases, especially as nix provides so many workarounds. Now that they are there I wont touch those but I feel we could have avoided that for eza/exa. Now I hope no one creates a new software called "exa" :s  

Anyway I dont feel strongly about this but just to show creating the aliases is not obvious for everyone.

**Pull request merged**